### ISSUE TITLE
Conflict with libconfig

### ISSUE THREAD

**Issue by Astaelan** [civetweb/civetweb#522]
libconfig is a popular C library used for configuration, while trying to incorporate civetweb into a project which uses libconfig, there is a conflict in the naming.
CONFIG_TYPE_STRING is already used in libconfig, I recommend a change of the naming prefix of all enumerations within civetweb.
A C library should use a very unlikely prefix, such as MG_CONFIG_TYPE_XXXX or CW_CONFIG_TYPE_XXXX.

**Comment by bel2125** [comment]
In principle, you are right - some prefix should be used.
But changing this now would be an incompatible interface change.

**Comment by Astaelan** [comment]
As with the other issue, I recommend using a LEGACY type define, so that the new version (1.10) going forward can address this issue but retain old compatibility for a while.

**Comment by bel2125** [comment]
V1.10 is going to be released now.
Some manual tests should be completed by now, so I don't want to change anything before the release anymore.

On the other hand, adding `MG_` now while keeping the versions without prefix, would not change any test result and would already allow using the prefixes.

**Comment by Astaelan** [comment]
Yep, so... I'm proposing to add it before 1.10 release so that it can be phased into mainstream sooner.  It should have no impact on any tests in the legacy operation.

**Comment by bel2125** [comment]
I agree. 
Still, for the current version you will need some define/undef for `CONFIG_TYPE_STRING`.
A legacy interface define will be set some time later in 1.11 or 1.12 - usually we allow for ~2 releases until some old interfaces are phased out.
But on the long term, every symbol will have this prefix.

**Comment by Astaelan** [comment]
For now I just prefixed all of the CONFIG_TYPE_XXX with MG_CONFIG_TYPE_XXX and maintain a repack of 1.9.1 on our CDN plus a mini download project to obtain it. When a civetweb stable release comes out with those changes, I'll adjust.
Thank you for the consideration on the symbols, will make things a little bit easier for everyone.

**Comment by bel2125** [comment]
I plan to release a V1.10 it with [these changes](https://github.com/civetweb/civetweb/commit/a01b3cdd06fba18225d5d94aa029ae3b7bbf73db#diff-61d374d2328431beee9d2b899fb64a2c) this week. You can already use the new symbols then.

Only removing the old ones will take some time (and you could either modify `civetweb.h` on your own or to some preprocessor magic in the meantime).

**Comment by Astaelan** [comment]
Are you going to update the source file with the default configuration array to use the new symbols?
RE:
`    {"cgi_pattern", MG_CONFIG_TYPE_EXT_PATTERN, "**.cgi$|**.pl$|**.php$"}, `
And all those that follow?

**Comment by bel2125** [comment]
Yes.
But only after the release.
Now I did only the minimum required to allow users to adapt to the new symbols as fast as possible.
Currently I don't want to do anything that could delay the release further. Anyway, I already planned to do a release almost a month ago, but then an update at Travis came with a very bad timing (#512) - so, currently I don't want any further delay (except "must have" bug fixes).
So, I will change this in ~2 weeks.

**Comment by Astaelan** [comment]
Understood, thank you.

**Comment by bel2125** [comment]
All symbols in the interface use `mg_`  or `MG_` as prefix now, the internal code is changed as well.
So *in principle* the symbols without prefix can be removed from the interface - but I will keep it for some versions for compatibility reasons. 

We could either close this issue, or keep it open as a remainder to remove the "old" symbols later and set it to "postponed".

**Comment by bel2125** [comment]
Postponed: Remove obsolete symbols without `mg_` prefix after some transition period.

**Comment by bel2125** [comment]
CONFIG_* without MG_ prefix are "phased out" by now.
